src/Filters/FiltersTrashed.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Spatie\QueryBuilder\Filters;
+
+use Illuminate\Database\Eloquent\Builder;
+
+/**
+ * FiltersTrashed provides filter for soft deleted (trashed) records.
+ *
+ * This filter responds to particular values:
+ *
+ * - 'with' - include 'trashed' records to the result set.
+ * - 'only' - return only 'trashed' records at the result set.
+ * - any other - return only records without 'trashed' at the result set.
+ *
+ * Usage example:
+ *
+ * ```php
+ * QueryBuilder::for(Item::class)
+ *     ->allowedFilters([
+ *         AllowedFilter::trashed(),
+ *         // ...
+ *     ]);
+ * ```
+ *
```

Comment:
Can we move these docs to the markdown files in the /docs directory? We haven't really got big doc blocks like this in the rest of the package.